solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2832,10 +2842,32 @@ public boolean checkIfCoreNodeNameAlreadyExists(CoreDescriptor dcore) {
    */
   public void publishNodeAsDown(String nodeName) {
     log.info("Publish node={} as DOWN", nodeName);
-    ZkNodeProps m = new ZkNodeProps(Overseer.QUEUE_OPERATION, OverseerAction.DOWNNODE.toLower(),
-        ZkStateReader.NODE_NAME_PROP, nodeName);
     try {
-      overseer.getStateUpdateQueue().offer(Utils.toJSON(m));
+      final AtomicBoolean sendToOverseer = new AtomicBoolean(false);
+      cc.getCoreDescriptors().parallelStream().forEach(cd -> {
+        DocCollection coll = zkStateReader.getCollection(cd.getCollectionName());
+        if (coll.isPerReplicaState()) {
+          final List<String> replicasToDown = new ArrayList<>();
+          coll.forEachReplica((s, replica) -> {
+            if (replica.getNodeName().equals(nodeName)) {
+              replicasToDown.add(replica.getName());
+            }
+          });
+          try {
+            PerReplicaStatesOps.downReplicas(replicasToDown,
+                PerReplicaStates.fetch(coll.getZNode(), zkClient, coll.getPerReplicaStates())).persist(coll.getZNode(), zkClient);
+          } catch (KeeperException | InterruptedException e) {
+            throw new RuntimeException(e);
+          }
+        } else {
+          sendToOverseer.set(true);
+        }
+      });
+      if (sendToOverseer.get()) {
```

Comment:
I think this change should 100% work 👍🏼 . 

Just want to throw out some thoughts here - would it be safer if we still send a DOWNNODE message to overseer? My minor concern is that overseer might in the future do more than just down replicas for DOWNNODE handling. In the logical sense, it might still be slightly safer if overseer is notified of a DOWNNODE event, just that it can choose to react to it or not for PRS collections, which is very clear also with the new change added in https://github.com/fullstorydev/lucene-solr/pull/183/files#diff-2cf13364680541a5642a57a6c3f16b565947f2ab70eed27a65942c2cb3a7e0bcR60 💯

Reply:
this looks suboptimal if we have multiple replicas of same collection in the node.

I believe we should collect a `Set<String>` of collections hosted by this node first and then we should identify the replicas for each collection that we need to remove from PRS

Reply:
Good call, I updated that to not repeat collections

Reply:
I also made the change to always send to overseer for node down and I added a comment in NodeMutator for why PRS collections are skipped in overseer. Hopefully that helps with stability in any future changes.